hub/core/serialize.py

```diff
@@ -312,10 +313,17 @@ def serialize_input_samples(
         buff = bytearray()
         nbytes = []
         shapes = []
+        expected_dim = len(meta.max_shape)
         for sample in samples:
             byts, shape = _serialize_input_sample(
                 sample, sample_compression, dtype, htype
             )
+            if htype == "image" and config.CONVERT_GRAYSCALE_IMAGES_TO_3D:
+                if not expected_dim:
+                    expected_dim = len(shape)
+                if len(shape) == 2 and expected_dim == 3:
+                    warnings.warn(f"Converting to 3D, 2D image with shape: {shape}")
```

Comment:
Can we word this better? Something like `Reshaping grayscale image with shape {shape} to {shape + (1,)} to match the dimensionality of the rest of the tensor`.

Reply:
Done.